inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLSource.java

```diff
@@ -0,0 +1,296 @@
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.inlong.agent.constant.TaskConstants.SQL_TASK_DATA_SEPARATOR;
+import static org.apache.inlong.agent.constant.TaskConstants.SQL_TASK_FETCH_SIZE;
+
+/**
+ * Read by SQL
+ */
+public class SQLSource extends AbstractSource {
+
+    public static final String AGENT_GLOBAL_SQL_SOURCE_PERMIT = "agent.global.sql.source.permit";
+    public static final int DEFAULT_AGENT_GLOBAL_SQL_SOURCE_PERMIT = 128 * 1000 * 1000;
+    private int MAX_RECONNECT_TIMES = 3;
+    private int RECONNECT_INTERVAL_SECOND = 10;
+    private int DEFAULT_FETCH_SIZE = 1000;
+
+    @Data
+    @AllArgsConstructor
+    @NoArgsConstructor
+    protected static class FileOffset {
+
+        private Long lineOffset;
+        private Long byteOffset;
+        private boolean hasByteOffset;
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SQLSource.class);
+    public static final String OFFSET_SEP = ":";
+    protected final Integer WAIT_TIMEOUT_MS = 10;
+    private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    private String fileName;
+    private String jdbcUrl;
+    private String username;
+    private String password;
+    private Connection conn;
+    protected BlockingQueue<SourceData> queue;
+    private static final ThreadPoolExecutor EXECUTOR_SERVICE = new ThreadPoolExecutor(
+            0, Integer.MAX_VALUE,
+            1L, TimeUnit.SECONDS,
+            new SynchronousQueue<>(),
+            new AgentThreadFactory("sql-source-pool"));
+    private volatile boolean running = false;
+    private boolean isMysql = true;
+    private volatile boolean fileExist = true;
+
+    public SQLSource() {
+    }
+
+    @Override
+    protected void initExtendClass() {
+        extendClass = DefaultExtendedHandler.class.getCanonicalName();
+    }
+
+    @Override
+    protected void initSource(InstanceProfile profile) {
+        try {
+            LOGGER.info("sql source init: {}", profile.toJsonStr());
+            AgentConfiguration conf = AgentConfiguration.getAgentConf();
+            int permit = conf.getInt(AGENT_GLOBAL_SQL_SOURCE_PERMIT, DEFAULT_AGENT_GLOBAL_SQL_SOURCE_PERMIT);
+            MemoryManager.getInstance().addSemaphore(AGENT_GLOBAL_SQL_SOURCE_PERMIT, permit);
+            fileName = profile.getInstanceId();
+            jdbcUrl = profile.get(TaskConstants.SQL_TASK_JDBC_URL).trim().replace("\r", "")
+                    .replace("\n", "");
+            if (jdbcUrl.startsWith("jdbc:mysql:")) {
+                isMysql = true;
+            }
```

Comment:
Does this require special handling for JDBC addresses such as `jdbc:hive` or `jdbc:clickhouse`?

Reply:
> Does this require special handling for JDBC addresses such as `jdbc:hive` or `jdbc:clickhouse`?

No, because setFetchSize only requires special handling in MySQL